logstash-core/lib/logstash/codecs/base.rb

```
@@ -7,8 +7,13 @@
 # This is the base class for logstash codecs.
 module LogStash::Codecs; class Base < LogStash::Plugin
   include LogStash::Config::Mixin
+
   config_name "codec"
 
+  def self.plugin_type
+    "codec"
```

This should raise a `NotImplementedError` instead of having a default.